src/libraries/System.Text.Json/tests/Serialization/Stream.Collections.cs

```diff
@@ -17,235 +18,247 @@ namespace System.Text.Json.Serialization.Tests
             }
             else if (type == typeof(ImmutableStack<TElement>))
             {
-                return ImmutableStack.CreateRange(GetArr_TypedElements(value));
+                return ImmutableStack.CreateRange(GetArr_TypedElements<TElement>(stringLength));
             }
             else if (type == typeof(ImmutableDictionary<string, TElement>))
             {
-                return ImmutableDictionary.CreateRange(GetDict_TypedElements(value));
+                return ImmutableDictionary.CreateRange(GetDict_TypedElements<TElement>(stringLength));
             }
             else if (
                 typeof(IDictionary<string, TElement>).IsAssignableFrom(type) ||
                 typeof(IReadOnlyDictionary<string, TElement>).IsAssignableFrom(type) ||
                 typeof(IDictionary).IsAssignableFrom(type))
             {
-                return Activator.CreateInstance(type, new object[] { GetDict_TypedElements(value) });
+                return Activator.CreateInstance(type, new object[] { GetDict_TypedElements<TElement>(stringLength) });
             }
             else if (typeof(IEnumerable<TElement>).IsAssignableFrom(type))
             {
-                return Activator.CreateInstance(type, new object[] { GetArr_TypedElements(value) });
+                return Activator.CreateInstance(type, new object[] { GetArr_TypedElements<TElement>(stringLength) });
             }
             else
             {
-                return Activator.CreateInstance(type, new object[] { GetArr_BoxedElements(value) });
+                return Activator.CreateInstance(type, new object[] { GetArr_BoxedElements<TElement>(stringLength) });
             }
         }
 
-        private static void AssertJsonEqual(string expected, string actual)
+        private static object GetEmptyCollection<TElement>(Type type)
         {
-            AssertJsonEqual(JsonDocument.Parse(expected).RootElement, JsonDocument.Parse(actual).RootElement);
+            if (type == typeof(TElement[]))
+            {
+                return Array.Empty<TElement>();
+            }
+            else if (type == typeof(ImmutableList<TElement>))
+            {
+                return ImmutableList.CreateRange(Array.Empty<TElement>());
+            }
+            else if (type == typeof(ImmutableStack<TElement>))
+            {
+                return ImmutableStack.CreateRange(Array.Empty<TElement>());
+            }
+            else if (type == typeof(ImmutableDictionary<string, TElement>))
+            {
+                return ImmutableDictionary.CreateRange(new Dictionary<string, TElement>());
+            }
+            else
+            {
+                return Activator.CreateInstance(type);
+            }
         }
 
-        private static void AssertJsonEqual(JsonElement expected, JsonElement actual)
+        private static string GetPayloadWithWhiteSpace(string json) => json.Replace(" ", new string(' ', 4));
+
+        private const int NumElements = 15;
```

Comment:
Reduced the number of collection elements to test for - this reduces the time to run the entire test suite from ~45s to ~28s, with no changes to code coverage.